### ISSUE TITLE
Firefox OS Simulator does not startup

### ISSUE THREAD

**Issue by javiersgithub** [mozilla/r2d2b2g#526]
First things first, i have downloaded the addon but it just doesn't starts and i have not found any error messages, so i can't really provide any useful information. Said that, i don't mind looking for better error messages i i'm told were to look.

The steps:
- Installed the stable 3 version from the add-on when doing a search in the integrated addon manager on firefox. This didn't start up (no new window open, no nothing, just a busy clock mouse pointer for a few seconds). 
- The i installed the 4.0pre1 simulator. Same result as before. Without any app loaded to it, that's just opening the simulator with no apps installed, the simulator does not start. What i'm able to see is the switch reading 'running' for a short amount of time (a seconds or so).
- I saw in this bug report https://github.com/mozilla/r2d2b2g/issues/380 a suggestion for enabling more detailed logs, unfortunately i didn't find that entry in about:config. I also explore the $HOME/.mozilla directory without luck.
- In the web console i only see the message: [01:19:46.658] Addon-message: isRunning, but it just shutdown after, and no more logs are displayed in the web console.

System details:
OS: Ubuntu Linux
Firefox: 21

**Comment by andharikarsanket** [comment]
apparently, it doesn't start with Private browsing enable. (dont know if this is intentional) disable the private browsing, restart firefox and it should work. you might need to switch back n forth every time if u want to use both the simulator and private browsing.

**Comment by javiersgithub** [comment]
Currently not using private mode.

I have been looking around into thee mozilla files and I'm not sure if what I have found is related or not, but at least is curious.
I have located the folder: $(HOME)/.mozilla/firefox/9x5m60gb.default/extensions/r2d2b2g@mozilla.org/resources/r2d2b2g/data/linux64/b2g, don't know is the executable file `b2g` is the actual simulator or not, but when trying to launch it i get several critical errors. The complete output:

./b2g 

(process:17209): GLib-CRITICAL **: g_slice_set_config: assertion `sys_page_size == 0' failed
Fontconfig warning: "/etc/fonts/conf.d/50-user.conf", line 14: reading configurations from ~/.fonts.conf is deprecated.
###!!! ABORT: Request 155.66: BadRequest (invalid request code or no such operation): file /home/myk/Mozilla/central/toolkit/xre/nsX11ErrorHandler.cpp, line 157
###!!! ABORT: Request 155.66: BadRequest (invalid request code or no such operation): file /home/myk/Mozilla/central/toolkit/xre/nsX11ErrorHandler.cpp, line 157
Segmentation fault (core dumped)

I really don't know who is requesting a cpp file and why (should this be already compiled?), but anyway seeing absolute paths with different user names it's suspicious.

If this is not the right way to launch the simulator (outside firefox, or even if that's not possible) let me know. Any hints to try to identify the issue are welcome.

Regards.

**Comment by javiersgithub** [comment]
Still doesn't work with firefox 22.

**Comment by mykmelez** [comment]
Hi @javiersgithub, I'm sorry you're having this problem. Here's some info to help diagnose it.

> - Installed the stable 3 version from the add-on when doing a search in the integrated addon manager on firefox. This didn't start up (no new window open, no nothing, just a busy clock mouse pointer for a few seconds).
> - The i installed the 4.0pre1 simulator. Same result as before. Without any app loaded to it, that's just opening the simulator with no apps installed, the simulator does not start. What i'm able to see is the switch reading 'running' for a short amount of time (a seconds or so).

I published a new preview build yesterday, so the next thing to try is installing the new build ([Windows](https://ftp.mozilla.org/pub/mozilla.org/labs/r2d2b2g/r2d2b2g-windows.xpi), [Mac](https://ftp.mozilla.org/pub/mozilla.org/labs/r2d2b2g/r2d2b2g-mac.xpi), and [Linux](https://ftp.mozilla.org/pub/mozilla.org/labs/r2d2b2g/r2d2b2g-linux.xpi)).

> - I saw in this bug report #380 a suggestion for enabling more detailed logs, unfortunately i didn't find that entry in about:config. I also explore the $HOME/.mozilla directory without luck.

The entry doesn't exist in about:config by default; you need to context-click on the configuration list to open its context menu and then select New > Integer.

> I have located the folder: $(HOME)/.mozilla/firefox/9x5m60gb.default/extensions/r2d2b2g@mozilla.org/resources/r2d2b2g/data/linux64/b2g, don't know is the executable file b2g is the actual simulator or not, but when trying to launch it i get several critical errors.

I'm not sure if these errors are significant, but in order to start the simulator from the command line, you need to pass it the location of the Gaia profile directory:

```
$(HOME)/.mozilla/firefox/9x5m60gb.default/extensions/r2d2b2g@mozilla.org/resources/r2d2b2g/data/linux64/b2g -profile $(HOME)/.mozilla/firefox/9x5m60gb.default/extensions/r2d2b2g@mozilla.org/profile
```

And if your Linux installation is 32-bit, then you need to start the 32-bit build of the simulator:

```
$(HOME)/.mozilla/firefox/9x5m60gb.default/extensions/r2d2b2g@mozilla.org/resources/r2d2b2g/data/linux/b2g -profile $(HOME)/.mozilla/firefox/9x5m60gb.default/extensions/r2d2b2g@mozilla.org/profile
```

**Comment by mykmelez** [comment]
@javiersgithub Have you had any luck with the latest preview build or enabling more detailed logs?

**Comment by javiersgithub** [comment]
I have done several tests but no luck. This time running Firefox 22.
- The same errors as before continue to appear, including the segment fault. 
- I have clone the repo an compile it my self, but same errors appear.
- Have installed the 32 bit compatibility libraries (my system is an Ubuntu 64) and now it does not crash and the toggle shows the 'running' label but apparently nothing happens.
- I was able to add some more details to the logs, but the important messages remain the same (the error ones).

I was looking to try to debug it ...., but i really don't know where to start :). I follow the Contributing file and was able to compile it and run it (crashing). Is the any easy way to debug it, is there any IDE's proyect file around that i didn't see, [Somewhere, in this repo if i recall correctly, i read debugging using gdb] is it easy to load the project to an IDE like eclipse, qtcretor, codeblocks or similar?

By the way, in the Contributing file in the 'Using Custom Profiles' section it refers to a step 9 from the previous section, which does not exists (there is a commit that did deleted steps 8 and 9, according to the comit's comment).

How often do you release a new version? and it is always available at the same url? or alternatively how do i create the xpi file from the cloned repo?.

Thanks for your interest.

**Comment by phillc73** [comment]
I was experiencing something similar, especially bullet point 3 in the post above. I thought my situation may shed some light on things. 

I started by installing v3.0 from the Mozilla Addon list, but have now updated to v4.0pre2.

I am running Crunchbang Waldorf 64-bit Linux. I have the 32-bit compatibility libraries installed. I am using Iceweasel v20.0.

FirefoxOS Simulator does not appear under the Tools/Web Developer menu. I can access the simulator directly at:

resource://r2d2b2g-at-mozilla-dot-org/r2d2b2g/data/content/index.html

When clicking "Stopped" it changes to "Running" (although text not clear) with an endless spiral cursor.

However, I have managed to launch the simulator successfully from the command line with the following:

<pre>
/home/phillc/.mozilla/firefox/b9xjwo41.default/extensions/r2d2b2g@mozilla.org/resources/r2d2b2g/data/linux64/b2g/b2g -profile /home/phillc/.mozilla/firefox/b9xjwo41.default/extensions/r2d2b2g@mozilla.org/profile
</pre>

So in short, the simulator does not work from within my browser, but can be launched via command line.

**Comment by javiersgithub** [comment]
Thanks for your advice @phillc73, but unfortunately I'm still not able to launch it, in-firefox neither standalone. This test has been done with the latest version, 4.0-pre3 released 2013-05-023 (yesterday). The console output is the same as always, but i provide them below just for reference.

javi@carbono:~/.mozilla/firefox/9x5m60gb.default/extensions/r2d2b2g@mozilla.org/resources/r2d2b2g/data/linux64/b2g$ ./b2g -profile /home/javi/.mozilla/firefox/9x5m60gb.default/extensions/r2d2b2g@mozilla.org/profile

(process:25380): GLib-CRITICAL **: g_slice_set_config: assertion `sys_page_size == 0' failed
Fontconfig warning: "/etc/fonts/conf.d/50-user.conf", line 14: reading configurations from ~/.fonts.conf is deprecated.
desktop-helper: +++ loading scripts for app: resource://gre-resources/hiddenWindow.html

###!!! ABORT: Request 155.66: BadRequest (invalid request code or no such operation): file /home/myk/Mozilla/central/toolkit/xre/nsX11ErrorHandler.cpp, line 157
###!!! ABORT: Request 155.66: BadRequest (invalid request code or no such operation): file /home/myk/Mozilla/central/toolkit/xre/nsX11ErrorHandler.cpp, line 157
Segmentation fault (core dumped)

**Comment by mykmelez** [comment]
I still can't reproduce this, but I have some additional hints.

First, there's andreasgal/B2G#210, which suggests that an incorrect value for B2G_HOMESCREEN could cause a problem like this. But the Simulator doesn't set that environment variable.

Then there's [Ubuntu bug 1160569](https://bugs.launchpad.net/ubuntu/+source/firefox/+bug/1160569), which tracks [Mozilla bug 833117](https://bugzilla.mozilla.org/show_bug.cgi?id=833117), which is a crash on startup due to an API change in the version of glib shipped with Ubuntu 13-04 Raring. The workaround for that is apparently to set `G_SLICE=always-malloc` in the environment, although one user reported that it didn't work for him.

Finally, there's [Ubuntu bug 1179554](https://bugs.launchpad.net/ubuntu/+source/glib2.0/+bug/1179554), which tracks [GNOME bug 674885](https://bugzilla.gnome.org/show_bug.cgi?id=674885). That causes a hang, however, rather than a crash; so it seems unlikely to be the cause of the problem.

**Comment by peshoicov** [comment]
I had - and still have - the same problem. When I see the console in FireBug, and click on the button to start the simulator - it says "Addon-message: isRunning".
So I managed to launch the simulator from the command line, as "phillc73" commented. Still searching for stable solution.
TechSpecs: OpenSUSE 12.3 64b, Gnome 3.6, Firefox 21.0, FirefoxOS Simulator 3.0.1.

**Comment by javiersgithub** [comment]
Hi, my situation has changed a lot for better, now it launchs!

As far as i know i might have suffer instabilities from other software. At first i thought that there weren't connected because firefox was running Ok, but after a regular software update the simulator start as it should. The problem is likely to be related to some library that the simulator uses that firefox doesn't, and that other software like VLC also uses ( VLC wasn't running before either). Some of the packages updated today were a new kernel and the proprietary amd graphics drivers (among others). 

Thanks for your efforts!. Now it's turn to play with it!

**Comment by mykmelez** [comment]
@javiersgithub I'm glad it works for you now!

@peshoicov I'm not sure what issue you're experiencing, but the next step is to try the latest preview build ([Windows](https://ftp.mozilla.org/pub/mozilla.org/labs/r2d2b2g/r2d2b2g-windows.xpi), [Mac](https://ftp.mozilla.org/pub/mozilla.org/labs/r2d2b2g/r2d2b2g-mac.xpi), [Linux](https://ftp.mozilla.org/pub/mozilla.org/labs/r2d2b2g/r2d2b2g-linux.xpi)). If that doesn't solve the problem, then let's open a new issue report for it, since it's different from the one reported here.

@phillc73 It sounds like your problem is actually #423, which is an Iceweasel issue.

**Comment by cutec-chris** [comment]
i think this is an opengl related issue. I have also drivers with very slow aceleration and have the same issue here.

**Comment by ghost** [comment]
OK, "Firefox OS Simulator does not  startup" está sendo resolvido agora.
Valeu pela informação para esse processo, Christian U.
Em 30/12/2013 12:30, "Christian U." notifications@github.com escreveu:

**Comment by ghost** [comment]
Nao conseguimos identificar sua resposta :-(  Talvez eu estou indisponivel.
Tente de novo! (0403)
Em 30/12/2013 12:30, "Christian U." notifications@github.com escreveu:
